unified-runtime/source/adapters/level_zero/sampler.cpp

```diff
@@ -88,16 +66,45 @@ ur_result_t urSamplerCreate(
       return UR_RESULT_ERROR_INVALID_VALUE;
     }
 
-    if (Props->filterMode == UR_SAMPLER_FILTER_MODE_NEAREST)
+    if (SamplerDesc->filterMode == UR_SAMPLER_FILTER_MODE_NEAREST)
       ZeSamplerDesc.filterMode = ZE_SAMPLER_FILTER_MODE_NEAREST;
-    else if (Props->filterMode == UR_SAMPLER_FILTER_MODE_LINEAR)
+    else if (SamplerDesc->filterMode == UR_SAMPLER_FILTER_MODE_LINEAR)
       ZeSamplerDesc.filterMode = ZE_SAMPLER_FILTER_MODE_LINEAR;
     else {
       UR_LOG(ERR, "urSamplerCreate: unsupported UR_SAMPLER_FILTER_MODE value");
       return UR_RESULT_ERROR_INVALID_VALUE;
     }
   }
 
+  return UR_RESULT_SUCCESS;
+}
+
+namespace ur::level_zero {
+
+ur_result_t urSamplerCreate(
+    /// [in] handle of the context object
+    ur_context_handle_t Context,
+    /// [in] specifies a list of sampler property names and their
+    /// corresponding values.
+    const ur_sampler_desc_t *Props,
+    /// [out] pointer to handle of sampler object created
+    ur_sampler_handle_t *Sampler) {
+  std::shared_lock<ur_shared_mutex> Lock(Context->Mutex);
+
+  // Have the "0" device in context to own the sampler. Rely on Level-Zero
+  // drivers to perform migration as necessary for sharing it across multiple
+  // devices in the context.
+  //
+  // TODO: figure out if we instead need explicit copying for acessing
+  // the sampler from other devices in the context.
+  //
+  ur_device_handle_t Device = Context->getDevices()[0];
+
+  ze_sampler_handle_t ZeSampler;
+  ZeStruct<ze_sampler_desc_t> ZeSamplerDesc;
+  ze_api_version_t ZeApiVersion = Context->getPlatform()->ZeApiVersion;
+  ur2zeSamplerDesc(ZeApiVersion, Props, ZeSamplerDesc);
```

Comment:
We should check the error value returned here in case `ur2zeSamplerDesc` doesn't return success.

Reply:
Missed to wrap in a `UR_CALL`  here. Good spot, thanks!